### PR TITLE
SALTO-5197: Added E2E for Assets

### PR DESCRIPTION
**Pull request by idoamit8**
In this PR I added E2E for all assets types

---

_Additional context for reviewer_

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/e2e_test/instances/cloud/index.ts [hunk 1]
```diff
@@ -234,6 +242,40 @@ export const createInstances = (
...
+  )
+
+  const objecttypeattribute = new InstanceElement(
+    `testSchema_Hardware_Assets_us_${randomString}@uumu`,
+    findType(OBJECT_TYPE_ATTRIBUTE_TYPE, fetchedElements),
+    createObjectTypeAttributeValues(randomString, fetchedElements),
+    undefined,
+  )
+  // eslint-disable-next-line no-console
+  console.log(`name are: ${objecttypeattribute.value.name}, ${objectType.value.name}, ${objectSchemaStatus.elemID.name}`)
```
Comment (shayc331) [inline packages/jira-adapter/e2e_test/instances/cloud/index.ts:277]: leftover?

File: packages/jira-adapter/e2e_test/instances/cloud/index.ts [hunk 2]
```diff
@@ -234,6 +242,40 @@ export const createInstances = (
...
+
+  const objectType = new InstanceElement(
+    `testSchema_Hardware_Assets_us_${randomString}@uumu`,
+    findType(OBJECT_TYPE_TYPE, fetchedElements),
+    createObjectTypeValues(randomString, fetchedElements),
+    undefined,
+    { [CORE_ANNOTATIONS.PARENT]: [objectSchemaRef] }
+  )
+
+  const objecttypeattribute = new InstanceElement(
```
Comment (shayc331) [inline packages/jira-adapter/e2e_test/instances/cloud/index.ts:270]: I think you meant: 
```suggestion
  const objectTypeAttribute = new InstanceElement(
```

File: packages/jira-adapter/src/config/api_config.ts [hunk 3]
```diff
@@ -2309,6 +2309,7 @@ const JSM_DUCKTYPE_TYPES: JiraDuckTypeConfig['types'] = {
         { fieldName: 'globalId' },
         { fieldName: 'objectSchemaId' },
         { fieldName: 'workspaceId' },
+        { fieldName: 'position' },
```
Comment (shayc331) [inline packages/jira-adapter/src/config/api_config.ts:2312]: Does it cause any noise?
Is it related to the e2e?

File: packages/jira-adapter/src/config/api_config.ts [hunk 4]
```diff
@@ -2349,6 +2350,7 @@ const JSM_DUCKTYPE_TYPES: JiraDuckTypeConfig['types'] = {
         { fieldName: 'globalId' },
         { fieldName: 'system' },
         { fieldName: 'referenceObjectType' }, // API returns referenceObjectTypeId as well.
+        { fieldName: 'position' },
```
Comment (shayc331) [inline packages/jira-adapter/src/config/api_config.ts:2353]: again, Does it cause any noise?

File: packages/jira-adapter/src/filters/assets/assets_instances_addition.ts [hunk 5]
```diff
@@ -44,6 +44,16 @@ const filter: FilterCreator = ({ config }) => ({
       .forEach(instance => {
         instance.value.objectSchemaId = getParent(instance).value.id
       })
+
+    await awu(changes)
```
Comment (shayc331) [inline packages/jira-adapter/src/filters/assets/assets_instances_addition.ts:48]: why do we add this part? 
can it be in a separate pr?

File: packages/jira-adapter/src/filters/assets/assets_instances_addition.ts [hunk 6]
```diff
@@ -44,6 +44,16 @@ const filter: FilterCreator = ({ config }) => ({
...
+      .filter(isInstanceChange)
+      .map(getChangeData)
+      .filter(instance => instance.elemID.typeName === OBJECT_TYPE_TYPE)
+      .forEach(instance => {
+        if (instance.value.parentObjectTypeId?.elemID.typeName === OBJECT_SCHEMA_TYPE) {
```
Comment (shayc331) [inline packages/jira-adapter/src/filters/assets/assets_instances_addition.ts:53]: nit: 
maybe use a filter instead of the if statement

File: packages/jira-adapter/src/filters/assets/assets_instances_addition.ts [hunk 7]
```diff
@@ -44,6 +44,16 @@ const filter: FilterCreator = ({ config }) => ({
       .forEach(instance => {
         instance.value.objectSchemaId = getParent(instance).value.id
       })
+
+    await awu(changes)
+      .filter(isInstanceChange)
+      .map(getChangeData)
```
Comment (shayc331) [inline packages/jira-adapter/src/filters/assets/assets_instances_addition.ts:50]: I think you should add: 
```suggestion
       .filter(isAdditionChange)
      .map(getChangeData)
```

**Review by shayc331 — COMMENTED** [review]
Looking great 🏆 
add some questions

File: packages/jira-adapter/e2e_test/instances/cloud/index.ts (hunk 1 above)
Comment (idoamit8) [reply packages/jira-adapter/e2e_test/instances/cloud/index.ts:277]: Yes, thanks :)

File: packages/jira-adapter/src/config/api_config.ts (hunk 3 above)
Comment (idoamit8) [reply packages/jira-adapter/src/config/api_config.ts:2312]: Yes, It is changing and the e2e helped me to catch it. 
It will not cause a noise reduction because its behind FF

File: packages/jira-adapter/src/config/api_config.ts (hunk 4 above)
Comment (idoamit8) [reply packages/jira-adapter/src/config/api_config.ts:2353]: Nope :)

File: packages/jira-adapter/src/filters/assets/assets_instances_addition.ts (hunk 5 above)
Comment (idoamit8) [reply packages/jira-adapter/src/filters/assets/assets_instances_addition.ts:48]: The E2E helped me understand that I cant deploy root objectTypes with this attribute, so I added it. I can do this in different PR but I think its ok to leave it here. it is very small change.

File: packages/jira-adapter/src/filters/assets/assets_instances_addition.ts (hunk 6 above)
Comment (idoamit8) [reply packages/jira-adapter/src/filters/assets/assets_instances_addition.ts:53]: agree! :)

File: packages/jira-adapter/src/filters/assets/assets_instances_addition.ts (hunk 7 above)
Comment (idoamit8) [reply packages/jira-adapter/src/filters/assets/assets_instances_addition.ts:50]: its true for modification change as well but I added it for those.

Comment (shayc331) [reply packages/jira-adapter/src/filters/assets/assets_instances_addition.ts:50]: so maybe you should add it in a different filter, or change the filter's name

Comment (idoamit8) [reply packages/jira-adapter/src/filters/assets/assets_instances_addition.ts:50]: you are correct. I changed the filter's name :)
Trying to reduce amount of filters.

File: packages/jira-adapter/src/filters/assets/assets_instances_deployment.ts [hunk 8]
```diff
@@ -44,6 +44,15 @@ const filter: FilterCreator = ({ config }) => ({
       .forEach(instance => {
         instance.value.objectSchemaId = getParent(instance).value.id
       })
+
+    await awu(changes)
+      .filter(isInstanceChange)
+      .map(getChangeData)
```
Comment (shayc331) [inline packages/jira-adapter/src/filters/assets/assets_instances_deployment.ts:50]: I think you should add it here as well
```suggestion
      .filter(isAdditionOrModificationChange)
      .map(getChangeData)
```

**Review by shayc331 — APPROVED** [review]
looking good!
add last comment, preApprove to save a cycle
